src/libs/ajax/loaded-data/useCachedData.ts

```diff
@@ -0,0 +1,31 @@
+import { Atom, LoadedState, ReadyState } from '@terra-ui-packages/core-utils';
+import { useLoadedDataEvents } from 'src/libs/ajax/loaded-data/useLoadedData';
+import { useSettableStore } from 'src/libs/react-utils';
+
+export const useCachedData = <S, HookResult extends LoadedState<S, unknown>>(
+  useHookInvoker: () => [HookResult, (dataCall: () => Promise<S>) => Promise<void>],
+  store: Atom<S>
+): [HookResult, (dataCall: () => Promise<S>) => Promise<void>] => {
+  const [stored, setStored] = useSettableStore(store);
+  const [dataResult, updateData] = useHookInvoker();
+  useLoadedDataEvents(dataResult, {
+    onSuccess: (result: ReadyState<S>) => {
+      setStored(result.state);
+    },
+  });
+
+  const finalResult =
```

Comment:
One unfortunate consequence of this is that the returned data object changes every render, which has consequences for downstream `useEffect`s. Not necessarily a blocker though.

Reply:
I've thought about this alot too... curious to pick your brain on this in a chat.

Reply:
Maybe a case for `useMemo`?

Reply:
`useMemo` is probably a good idea, given that the potential of the data changing triggering other network calls. 
Particularly given that I'd like us to move _towards_ using effect hooks with proper dependency arrays, and _away_  from `useOnMount`.

Reply:
agreed @blakery , tho this PR is getting long in the tooth... it makes things no worse then before, so let's tackle the memo optimization as a followup PR.